xdsl/dialects/smt.py

```diff
@@ -196,6 +197,29 @@ class XOrOp(VariadicBoolOp):
     name = "smt.xor"
 
 
+@irdl_op_definition
+class ImpliesOp(IRDLOperation):
```

Comment:
Am I missing something, why not use `VariadicBoolOp`? The docstring you gave even says this takes a variadic number of operands, though the definition does not agree with this

Reply:
I need to fix the docstring, my bad

Reply:
Done!